src/test/java/lambda/part3/exercise/ReachIterableTest.java

```diff
@@ -44,24 +55,21 @@
         }
 
         default boolean anyMatch(final Predicate<T> predicate) {
-            final boolean[] isAnyMatched = { false };
-            final Predicate<T> isMatched = element -> isAnyMatched[0] = predicate.test(element);
-            while (!isAnyMatched[0] && this.forNext(isMatched::test));
-            return isAnyMatched[0];
+            final boolean[] matched = { false };
+            final Consumer<T> testAndExtract = element -> matched[0] = predicate.test(element);
+            while (!matched[0] && this.forNext(testAndExtract));
+            return matched[0];
         }
 
         default boolean allMatch(final Predicate<T> predicate) {
-            final boolean[] isAllMatched = { true };
-            final Predicate<T> isMatched = element -> isAllMatched[0] = predicate.test(element);
-            while (isAllMatched[0] && this.forNext(isMatched::test));
-            return isAllMatched[0];
+            final boolean[] matched = { true };
```

Comment:
Couldn't understand how implement it without private methods in interface or state. Should i use something tricky?